### PR TITLE
Master+atsamd21 xpro

### PR DESCRIPTION
**Pull request by 0xB0D**
Hello.

This PR enables Zephyr on the SAM D21 Xplained Pro. The base footprint of the D20 Xplained and D21 Xplained is the same, however, the SAMD21 has a different pinout and different I/O connections as a result and therefore requires a new board port to enable Zephyr.

The first patch fixes a simple compile error
The second patch lands the new board port.

Note I'm open to breaking up the second patch into smaller components, I'm landing it as one patch here since that is how the SAM D20 landed, so I'm assuming you guys prefer it that way.

If not let me know and I can churn into smaller pieces.

### CODE REVIEW

**Comment by bryanodonoghue** [comment]
I had this error with gitlint too

Commit bbeb3f2e0c:
1: UC2 Signed-off-by: must have a full name

But my SOB looks like this for both patches
    Signed-off-by: Bryan O'Donoghue <pure.logic@nexus-software.ie>

I guess it's the apostrophe - which is common for Irish names BTW.

**Comment by bryanodonoghue** [comment]
Confirmed gitlint wants "Bryan ODonoghue" not "Bryan O'Donoghue" ... my given name.
Can you guys commit as-is without having to change my SOB ?

File: boards/arm/atsamd21_xpro/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+if(CONFIG_PINMUX_SAM0)
```
Comment (galak) [inline boards/arm/atsamd21_xpro/CMakeLists.txt]: Can you add copyright/license header info

File: boards/arm/atsamd21_xpro/atsamd21_xpro_defconfig [hunk 2]
```diff
@@ -0,0 +1,21 @@
+CONFIG_ARM=y
```
Comment (galak) [inline boards/arm/atsamd21_xpro/atsamd21_xpro_defconfig]: Can you add copyright/license header info

**Review by galak — CHANGES_REQUESTED** [review]
A few copyright/license headers missing.  Otherwise at first glance this looks good.

**Comment by galak** [comment]
> Confirmed gitlint wants "Bryan ODonoghue" not "Bryan O'Donoghue" ... my given name.
> Can you guys commit as-is without having to change my SOB

Yeah, we should open an issue to get gitlint fixed not to flag that.

Can you see if you can address the other commit message (I think it was a little too long).

**Comment by galak** [comment]
@dbkinder can you take a look at the docs.

File: boards/arm/atsamd21_xpro/CMakeLists.txt (hunk 1 above)
Comment (bryanodonoghue) [reply boards/arm/atsamd21_xpro/CMakeLists.txt]: done

File: boards/arm/atsamd21_xpro/atsamd21_xpro_defconfig (hunk 2 above)
Comment (bryanodonoghue) [reply boards/arm/atsamd21_xpro/atsamd21_xpro_defconfig]: done

**Review by dbkinder — APPROVED** [review]
LGTM, thanks!

**Comment by galak** [comment]
@bryanodonoghue once you push the other fixes we should be able to merge this!

**Review by galak — APPROVED** [review]

**Pull request merged**